methods/did:key.md

```diff
@@ -0,0 +1,47 @@
+# did:key
+
+The [Key DID method](https://w3c-ccg.github.io/did-method-key) encoding leverages the existing multicodes used by the Key DID method.
```

Comment:
My main problem is that unlike https://github.com/ucan-wg/ucan-ipld/#21-principal we have to tag keys with additional `0x0d1d` umbrella tag, rendering all our UCANs invalid.

This may not be a deal-breaker necessarily, however I'm not entirely convinced this approach is strictly better. While it more explicitly signals that value is a `did:key` as opposed to just public key, I would argue that both are just public keys formatted differently and given the context they are used in it's obvious which one is implied.

While it may be more challenging to specify multidid without umbrella tag, I still think it may be a good idea because that way one could go and implement that spec, while approach with  `<multidid-code><method-code>` is practically impossible as there could be some unknown `method-code` making it unclear how to interpret method-bytes.

Reply:
A few problems I've observed with not having a `multidid-code`:
* How do I know what I'm looking at is a DID and should be treated as such? (e.g. just a public key or a DID key)
* In the new version of 3ID we are planning to use a multihash as the identifier, in this case how would I know that I'm looking at a multihash vs a DID?
* For DID PKH the plan is to use `0xca`, this is already registered as "CAIP-50 multi-chain account id", not as DID

> is practically impossible as there could be some unknown method-code making it unclear how to interpret method-bytes.

How is this different from just having a bunch of different codes for each DID method? At least with a DID umbrella tag the implementation can know it's a DID and throw an appropriate error (e.g. method not supported)

Reply:
Oh, and the main thing that we wouldn't get without the `0x0d1d` tag is a way to encode an entire DID URL.

For example you can specify a specific verification method on DID key by:
```
did:key:z6MkiTBz1ymuepAQ4HEHYSF1H8quG5GLVVQR3djdX3mDooWp#z6LShs9GGnqk85isEBzzshkuVWrVKsRp24GnDuHk8QWkARMW
```
Here, we refer to the x25519 Verification Method of a ed25519 Key DID.

Reply:
Ok, I think I'm ok with multidid requiring `0x0d1d` and in contexts where it's clear what you mean from the field name like `iss` and `aud` in UCANs we can define them as union of MultiDID and DIDKey

Reply:
> in UCANs we can define them as union of MultiDID and DIDKey

I understand why you'd want to support reading a DIDKey without the multidid code for backwards compatibility. I assume that all new writes would still use MiltiDID? Otherwise we might end up with different ways of writing the same capability based on implementation?

Reply:
I need to think more about it. We have observed similar tension in [UCAN invocation](https://github.com/ucan-wg/invocation/pull/10) spec, where we need to tag invocations in context where you may be allowing others things and do not tag where only thing you allow is invocations.

If we end up allowing [non DID principals in UCANs as discussed here](https://github.com/ucan-wg/spec/issues/139) I absolutely agree that prefixing with `0x0d1d` is to go. On the other hand if we only allow DIDs I would argue that canonical representation should probably omit the tag.

That said I don't think it should affect this spec any way though.

Reply:
> If we end up allowing https://github.com/ucan-wg/spec/issues/139 I absolutely agree that prefixing with 0x0d1d is to go.

Interesting point. In this case it might be worth adding the `0x0d1d` prefix regardless for future compatibility in case non DID Principals become a thing?